pycloudlib/openstack/cloud.py

```diff
@@ -78,8 +78,12 @@ def get_instance(self, instance_id) -> OpenstackInstance:
         return OpenstackInstance(
             key_pair=self.key_pair,
             instance_id=instance_id,
+            network_id=self._get_network_id()
         )
 
+    def _get_network_id(self):
+        return self.conn.network.find_network(self.network).id
```

Comment:
Can `find_network` fail ? And if it can, we could have a better error message here instead of probably `NoneType has no attribute named id`